src/Core/FakeVimProxy.cpp

```diff
@@ -0,0 +1,491 @@
+/*
+ * Copyright (C) 2019-2020 Ashar Khan <ashar786khan@gmail.com>
+ *
+ * This file is part of CP Editor.
+ *
+ * CP Editor is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * I will not be responsible if CP Editor behaves in unexpected way and
+ * causes your ratings to go down and or lose any important contest.
+ *
+ * Believe Software is "Software" and it isn't immune to bugs.
+ *
+ */
+
+/* Most of the code in this file is under the following copyright
+
+    Copyright (c) 2017, Lukas Holecek <hluk@email.cz>
+    This file is part of CopyQ.
+    CopyQ is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+    CopyQ is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+    You should have received a copy of the GNU General Public License
+    along with CopyQ.  If not, see <http://www.gnu.org/licenses/>.
+
+*/
+#include "Core/FakeVimProxy.hpp"
+#include "Core/EventLogger.hpp"
+#include "appwindow.hpp"
+#include "fakevimactions.h"
+#include "fakevimhandler.h"
+#include "generated/SettingsHelper.hpp"
+#include "mainwindow.hpp"
+#include <QApplication>
+#include <QDebug>
+#include <QLabel>
+#include <QMainWindow>
+#include <QMessageBox>
+#include <QPlainTextEdit>
+#include <QStatusBar>
+#include <QTemporaryFile>
+#include <QTextBlock>
+#include <QTextEdit>
+#include <QTextStream>
+#include <QWidget>
+
+namespace Core
+{
+
+using _ = QLatin1String;
+FakeVimProxy::FakeVimProxy(QWidget *widget, MainWindow *mw, AppWindow *aw, QObject *parent)
```

Comment:
FakeVim Proxy takes an AppWindow instance to invoke actions slots like that which opens preference window, new file, format and others.